### PR TITLE
Jakub Lipowski, 209378

### PR DESCRIPTION
**Pull request by Dragoonov**

### CODE REVIEW

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java [hunk 1]
```diff
@@ -1,33 +1,67 @@
...
 
-	public int total = -1;
+	private final int[] numbers = new int[STACK_CAPACITY];
 
-	public void countIn(int in) {
+	/**
+	 * Gets total amount of numbers placed on stack minus 1
+	 * @return total amount of numbers placed on stack minus 1
+	 */
+	public int getTotal() {
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java:17]: field used only internally - should not have a getter

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java [hunk 2]
```diff
@@ -1,33 +1,67 @@
...
+		return total;
+	}
+
+	private int total = EMPTY_STACK_VALUE;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java:21]: Extracting this constant is done incorrectly. Should show during next classes ;)

File: src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+}
+
+
 public class IntLinkedList {
 
-	Node last;
-	int i;
+	private static final int EMPTY_VALUE = -1;
+	private Node last;
+	private final int i=0;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java:39]: Unused private field could be removed.

**Review by kowallus — COMMENTED** [review]
5,5 pts

Specific issues inlined in the code.

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java (hunk 1 above)
Comment (Dragoonov) [reply src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java:17]: There is an instruction to generate getter for total. 3.1.7

**Pull request closed without merging**